File: Data Structures & Algorithms/house-robber/submission-0.py
class Solution:
    def rob(self, nums: List[int]) -> int:

        prev2  = 0
        prev = nums[0]



        for i in range(1, len(nums)):


            take = nums[i] + prev2

            nottake = prev


            curr = max(take, nottake)


            prev2 = prev
            prev = curr

        return prev

            



        

File: Data Structures & Algorithms/jump-game/submission-0.go
func max(a, b int) int {
	if a > b {
		return a
	}

	return b
}


func canJump(nums []int) bool {

	var maxInd int


	for i:=0; i < len(nums); i++{

		if i > maxInd {
			return false
		}

		maxInd = max(maxInd, nums[i] + i)


	}
	
	return true
}
